executor/distsql.go

```diff
@@ -888,6 +891,7 @@ func (e *XSelectTableExec) Next() (*Row, error) {
 		}
 		if rowData == nil {
 			// Finish the current partial result and get the next one.
+			e.partialResult.Close()
```

Comment:
But in this case, the result has been consumed already, there should not have memory leak.